### ISSUE TITLE
[deployer] Make all processors fail execution attribute configurable

### ISSUE THREAD

**Issue by russdanner** [craftercms/craftercms#4033]
**Is your feature request related to a problem? Please describe.**
Today most out of the box processors fail execution attribute is hard coded to true when it should be configurable based on the deployment's needs.  

**Describe the solution you'd like**
* Provide a way for a processor to define a sensible default if no configuration is set 
* Allow a value to be set via YAML to override the configuration.

**Describe alternatives you've considered**
None

**Comment by yacdaniel** [comment]
@jdrossl I added a processor
```
      - processorName: commandLineProcessor
        workingDir: '/Users/yacdaniel'
        command: './script.sh'
        failDeploymentOnFailure: true
```
and the content of the script is 
```
#!/bin/bash
echo hello tests
exit -1
```

**Comment by jdrossl** [comment]
The issue can't be reproduced

**Comment by yacdaniel** [comment]
Verified in `Crafter CMS v3.1.8-SNAPSHOT`

**Comment by yacdaniel** [comment]
Verified fixed in `Crafter CMS v4.0.0-SNAPSHOT`
